gen_template.rb

```diff
@@ -82,6 +82,29 @@ def gen_template(
           ['-m', '{{ user `memory` }}'],
         ],
       ),
+      builder(
+        type: 'hyperv-iso',
+        generation: 1,
+        # generation: 2, # gen 2 fails uefi boot, missing configuration in builders/hyperv-iso.nix
+        iso_url: iso_url,
+        iso_checksum: iso_sha256,
+        boot_wait: "60s",
+        boot_command: [
+          "echo http://{{ .HTTPIP }}:{{ .HTTPPort}} > .packer_http<enter>",
+          "mkdir -m 0700 .ssh<enter>",
+          "curl $(cat .packer_http)/install_rsa.pub > .ssh/authorized_keys<enter>",
+          "sudo nix-env -f '<nixpkgs>' -iA linuxPackages.hyperv-daemons<enter><wait10>",
+          "sudo $(find /nix/store -executable -iname 'hv_kvp_daemon' | head -n 1)<enter><wait10>",
```

Comment:
Not sure if there's a better way to determine to path to `hv_kvp_daemon`; this daemon needs to be running during live-cd installation in order to pass back the dynamically assigned guest-ip back to the host (in order for packer to continue it's setup over ssh).

Reply:
worth note: went this direction, over `nixos-rebuild switch` and setting `virtualisation.hypervGuest.enable` in `configuration.nix`, because the execution time is greater reduced.

Reply:
You can get the store path with `nix-build --expr "(import <nixpkgs> {}).linuxPackages.hyperv-daemons"` for example. But I think it's fine like that. The chances that there are two versions of the same binary is probably fairly low.

Reply:
![image](https://user-images.githubusercontent.com/4636752/80325704-d814bc00-87fb-11ea-8d6f-0517f5d92e10.png)
![image](https://user-images.githubusercontent.com/4636752/80325708-dc40d980-87fb-11ea-8764-80a40a9e8385.png)

The two different methods resolved to different paths in the nixstore, seems to have to do with `hyperv-daemons` vs `hyperv-daemons-bin`.

Not exactly sure how to properly access the latter however,  `nix-build --expr "(import <nixpkgs> {}).linuxPackages.hyperv-daemons-bin"` does not work. So stuck with the original method using `find`. Would love to know what the best way is here, but ultimately agree;  chances that there are two versions of the same binary is low.